src/pages/index.js

```diff
@@ -85,125 +141,133 @@ const cardUrlInput = addCardFormElement.querySelector(".modal__input_type_url");
 
 // Functions
 
-// function closeModal(modal) {
-//   modal.classList.remove("modal_opened");
-//   document.removeEventListener("keydown", handleEscape);
-//   modal.removeEventListener("mousedown", handlePopupClose);
-// }
-
-// function openModal(modal) {
-//   modal.classList.add("modal_opened");
-//   document.addEventListener("keydown", handleEscape);
-//   modal.addEventListener("mousedown", handlePopupClose);
-// }
-
 function handleImageClick(cardData) {
-  // modalImage.src = cardData.link;
-  // modalImage.alt = cardData.name;
-  // modalTitle.textContent = cardData.name;
   imagePreviewPopup.open(cardData);
 }
 
-// function renderCard(cardData) {
-//   const cardElement = createCard(cardData);
-//   cardSection.addItem(cardElement);
-// }
-
-// function getCardElement(cardData) {
-//   const cardElement = cardTemplate.cloneNode(true);
-//   const cardImageElement = cardElement.querySelector(".card__image");
-//   const cardTitleElement = cardElement.querySelector(
-//     ".card__description-title"
-//   );
-//   const likeButton = cardElement.querySelector(".card__like-button");
-
-//   likeButton.addEventListener("click", () => {
-//     likeButton.classList.toggle("card__like-button_active");
-//   });
-
-//   const trashButton = cardElement.querySelector(".card__trash-button");
-//   trashButton.addEventListener("click", () => {
-//     cardElement.remove();
-//   });
-
-//   cardImageElement.addEventListener("click", () => {
-//     modalImage.src = cardData.link;
-//     modalImage.alt = cardData.name;
-//     modalTitle.textContent = cardData.name;
-//     imagePreviewPopup.open();
-//   });
-
-//   cardImageElement.src = cardData.link;
-//   cardTitleElement.textContent = cardData.name;
-//   cardImageElement.alt = cardData.name;
-//   return cardElement;
-// }
-
 function handleProfileEditSubmit(data) {
-  profileTitle.textContent = profileTitleInput.value;
-  profileDescription.textContent = profileDescriptionInput.value;
-  editProfilePopup.close();
+  // data is an object with name and subtitle
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```